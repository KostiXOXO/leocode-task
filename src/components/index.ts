import UsersList from './list/UsersList';
import SearchBox from './searchBox/SearchBox';
import FiltrableUsersList from './filtrableUsersList/FiltrableUsersList';
import Spinner from './spinner/Spinner';

export { UsersList, SearchBox, FiltrableUsersList, Spinner };
