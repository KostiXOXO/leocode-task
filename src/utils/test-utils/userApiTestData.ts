export const sampleData = [
	{
		id: 1,
		name: 'test user',
		username: 'test_user1',
	},
	{
		id: 2,
		name: 'tetrisuser',
		username: 'noneman',
	},
];
