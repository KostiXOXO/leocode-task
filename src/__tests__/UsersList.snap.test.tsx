import React from 'react';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { UsersList } from 'components';
import { sampleData } from 'utils/test-utils/userApiTestData';

afterEach(cleanup);

it('Component did not change', () => {
	const tree = renderer.create(<UsersList items={sampleData} />).toJSON();
	expect(tree).toMatchSnapshot();
});
