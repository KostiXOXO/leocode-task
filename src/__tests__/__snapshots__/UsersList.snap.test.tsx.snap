// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component did not change 1`] = `
<ul
  className="listContainer unordered"
  data-testid="listComponent"
>
  <li>
    <div>
      <span
        className="name"
      >
        <strong>
          test user
        </strong>
      </span>
      <span
        className="username"
      >
        @test_user1
      </span>
    </div>
  </li>
  <li>
    <div>
      <span
        className="name"
      >
        <strong>
          tetrisuser
        </strong>
      </span>
      <span
        className="username"
      >
        @noneman
      </span>
    </div>
  </li>
</ul>
`;
